src/mds/ScrubStack.cc

```diff
@@ -528,6 +528,52 @@ bool ScrubStack::scrub_in_transition_state() {
   return false;
 }
 
+std::string ScrubStack::scrub_summary() {
+  ceph_assert(mdcache->mds->mds_lock.is_locked_by_me());
+
+  bool have_more = false;
+  std::stringstream ss;
```

Comment:
These are rather expensive to construct. Let's instead use `CachedStackStringStream`.

Reply:
ack